scripts/generate_search_index.py

```
@@ -58,62 +58,110 @@ def is_arabic_file(path):
+                    if metadata.get("arTitle"):
+                        manifest_entry["arTitle"] = metadata.get("arTitle")
+
+                    if language != "ar":
+                        manifest_files.append(manifest_entry)
 
-                print(f"Indexed: {rel_path} ({language})")
+                    print(f"Processed: {rel_path} ({language})")
 
-            except Exception as e:
-                print(f"Error processing {rel_path}: {e}")
+                except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)